tests/trainer/test_trainer.py

```diff
@@ -664,6 +665,118 @@ def test_benchmark_option(benchmark_, deterministic, expected):
     torch.backends.cudnn.benchmark = original_val
 
 
+@pytest.mark.parametrize("ckpt_path", (None, "last"))
+@pytest.mark.parametrize("save_last", (True, False))
+@pytest.mark.parametrize("fn", ("fit", "validate"))
+@pytest.mark.parametrize("ft_enabled", (True, False))
+def test_checkpoint_path_input_last(tmpdir, ckpt_path, save_last, fn, ft_enabled):
```

Comment:
there is a bit much going on in this test. The highly nested if-else blocks make the test hard to follow and make the parametrizations almost pointless. 
I would at least disentangle the the fault tolerance from this test.

Reply:
Cleaned up - now we have separate test for fault tolerancy enabled